### PR TITLE
Invoker type tweaks

### PR DESCRIPTION
**Pull request by psuter**
Replaces type aliases with case classes when appropriate.

Other minor Scala tweaks (e.g. eliminating procedure syntax).

### CODE REVIEW

File: core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 1]
```diff
@@ -261,16 +265,19 @@ class ContainerPool(
      */
     def retrieve(key: String)(implicit transid: TransactionId): ContainerResult = {
         this.synchronized {
-            if (activeCount() + startingCounter.cur >= _maxActive)
-                return Busy()
```
Comment (markusthoemmes) [inline core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala]: \*gasp\*

**Comment by psuter** [comment]
PG1 281

**Comment by psuter** [comment]
FYI @rabbah

**Comment by psuter** [comment]
PG2 10

**Comment by psuter** [comment]
PG2 11 for confirmation

File: core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 2]
```diff
@@ -282,14 +291,15 @@ class ContainerPool(
...
             assert(ci.state == State.Active)
             assert(keyMap.contains(oldKey))
-            if (!keyMap.contains(newKey))
+            if (!keyMap.contains(newKey)) {
                 keyMap += newKey -> new ListBuffer()
-            val oldBucket = keyMap.get(oldKey).getOrElse(null)
-            val newBucket = keyMap.get(newKey).getOrElse(null)
+            }
+            val oldBucket = keyMap(oldKey)
+            val newBucket = keyMap(newKey)
```
Comment (markusthoemmes) [inline core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala]: Guaranteed to be set? Will throw an Exception if not.

**Comment by psuter** [comment]
Was merged as e30f8340e2e1c45fb2a397eb299f0318cdd92053

**Pull request closed without merging**